ts/src/gemini.ts

```diff
@@ -271,6 +278,123 @@ export default class gemini extends Exchange {
         });
     }
 
+    async fetchCurrencies (params = {}) {
+        /**
+         * @method
+         * @name gemini#fetchCurrencies
+         * @description fetches all available currencies on an exchange
+         * @param {object} params extra parameters specific to the endpoint
+         * @returns {object} an associative dictionary of currencies
+         */
+        return await this.fetchCurrenciesFromWeb (params);
```

Comment:
@ttodua we were trying to avoid doing this because the users had a lot of network errors when we were getting the markets by scraping the web page

Reply:
I have added `option["fetchCurrencies"]["enabled"] = true/false` with an exception message (in case of failure) telling user that s/he can disable that method. I think it will be good now. if 9 of 10 cases work and just rarely throws exceptions, users would be still ok with it, as it grabs important info (which is otherwise not possible to get even from gemini api), and in case of unexpected behavior, there is now additional option to disable it.

Reply:
wait, I am adding a more readable & better maintainable way for this. will comment soon

Reply:
I've updated. Now the code is more cleaner in gemini and with re-usable method (in other exchanges we have similarly hardcoded parts, so we can wrap them too in more cleaner/standardized route)